src/main/java/ch/njol/skript/expressions/ExprKeptItems.java

```diff
@@ -0,0 +1,140 @@
+ *
+ * Copyright Peter Güttinger, SkriptLang team and contributors
+ */
+package ch.njol.skript.expressions;
+
+import java.util.List;
+
+import org.bukkit.event.Event;
+import org.bukkit.event.entity.PlayerDeathEvent;
+import org.bukkit.inventory.ItemStack;
+import org.eclipse.jdt.annotation.Nullable;
+
+import ch.njol.skript.Skript;
+import ch.njol.skript.aliases.ItemType;
+import ch.njol.skript.classes.Changer.ChangeMode;
+import ch.njol.skript.doc.Description;
+import ch.njol.skript.doc.Events;
+import ch.njol.skript.doc.Examples;
+import ch.njol.skript.doc.Name;
+import ch.njol.skript.doc.RequiredPlugins;
+import ch.njol.skript.doc.Since;
+import ch.njol.skript.lang.Expression;
+import ch.njol.skript.lang.ExpressionType;
+import ch.njol.skript.lang.SkriptParser.ParseResult;
+import ch.njol.skript.lang.util.SimpleExpression;
+import ch.njol.util.Kleenean;
+import ch.njol.util.coll.CollectionUtils;
+
+@Name("Kept Items")
+@Description({"The kept items in a 'death of player' event.",
+	"",
+	"Note: this expression won't remove the items from the drops when changed."})
+@Examples({
+	"on death of player:",
+	"\tset kept items to the drops where [input isn't victim's tool]"})
+@Events("death")
+@RequiredPlugins("Paper 1.13+")
+@Since("INSERT VERSION")
+public class ExprKeptItems extends SimpleExpression<ItemType> {
+
+	static {
+		if (Skript.methodExists(PlayerDeathEvent.class, "getItemsToKeep"))
+			Skript.registerExpression(ExprKeptItems.class, ItemType.class, ExpressionType.SIMPLE, "[the] kept items");
+
+	}
+
+	@Override
+	public boolean init(Expression<?>[] exprs, int matchedPattern, Kleenean isDelayed, ParseResult parseResult) {
+		if (!getParser().isCurrentEvent(PlayerDeathEvent.class)) {
+			Skript.error("'kept items' can only be used in a 'death of player' event.");
+			return false;
+		}
+		return true;
+	}
+
+	@Override
+	protected @Nullable ItemType[] get(Event event) {
+		return ((PlayerDeathEvent) event).getItemsToKeep()
+			.stream()
+			.map(ItemType::new)
+			.toArray(ItemType[]::new);
+	}
+
+	@Override
+	public @Nullable Class<?>[] acceptChange(ChangeMode mode) {
+		switch (mode) {
+			case ADD:
+			case REMOVE:
+			case REMOVE_ALL:
+			case DELETE:
+			case RESET:
+				return CollectionUtils.array(ItemType.class);
+			default:
+				return null;
+		}
+	}
+
+	@Override
+	public void change(Event event, @Nullable Object[] delta, ChangeMode mode) {
+		if (!(event instanceof PlayerDeathEvent))
+			return;
+
+		List<ItemStack> keptItems = ((PlayerDeathEvent) event).getItemsToKeep();
+
+		// Clear items before adding the new values
+		if (mode == ChangeMode.SET)
+			keptItems.clear();
+
+		for (Object o : delta) {
+			ItemType item = (ItemType) o;
+			switch (mode) {
+				case ADD:
+				case SET:
+					item.addTo(keptItems);
+					break;
+				case REMOVE:
+					item.removeFrom(keptItems);
+					break;
+				case REMOVE_ALL:
+					item.removeAll(keptItems);
+				case DELETE:
+				case RESET:
+					keptItems.clear();
+					break;
```

Comment:
These cases would cause an exception because delta would always be null and you can't iterate though a null array.

Reply:
They wouldn't cause an exception as for loops whose looped array is null just don't run at all however you are right, this wouldn't work properly

Reply:
Looping any null object would throw an exception. Whether it was an array or an object implementing `Iterator`.